### PR TITLE
Componentize and style

### PR DESCRIPTION
**Pull request by jkruse14**
broke the editor into components and added a _styles_ prop to each to pass down styles allowing the user to customize the editor.

### CODE REVIEW

**Comment by jrm2k6** [comment]
Sweet. Looks like the unit tests are broken though. I should be able to review that today or tomorrow. Thanks a lot!

**Comment by jkruse14** [comment]
Ah, I know what it is. I linked this to another project with npm link. Since your travis script installs react with each build, I was getting an error due to having duplicate reacts in my node_modules (something funky with linking, I haven't quite found the answer yet). So, I added a line to remove react in the travis before script, so now the tests cannot find react... Sorry! Completely forgot about that!

File: src/MarkdownEditor.js [hunk 1]
```diff
@@ -302,6 +64,10 @@ var MarkdownEditor = React.createClass({
       'position': 'relative'
     };
 
+    if(this.props.styles.styleMarkdownEditorHeader) {
```
Comment (jrm2k6) [inline src/MarkdownEditor.js]: just a missing space between if and opening parenthesis

File: src/MarkdownEditor.js [hunk 2]
```diff
@@ -44,12 +41,28 @@ var MarkdownEditor = React.createClass({
   render: function() {
     var divContent;
     var editorMenu;
+    var styles;
+    if(!this.props.styles) {
```
Comment (jrm2k6) [inline src/MarkdownEditor.js]: Would it be better to use [defaultProps?](https://facebook.github.io/react/docs/react-component.html#defaultprops) 

Just add some spaces around the colon and the empty object, and between the if and opening parenthesis

File: src/MarkdownEditor.js [hunk 3]
```diff
@@ -65,7 +78,7 @@ var MarkdownEditor = React.createClass({
     };
 
     if(this.props.styles.styleMarkdownEditorHeader) {
-      styleMarkdownEditorHeader = this.props.styles.styleMarkdownEditorHeader;
+      Object.assign(styleMarkdownEditorHeader, this.props.styles.styleMarkdownEditorHeader);
```
Comment (jrm2k6) [inline src/MarkdownEditor.js]: Is it backward compatible? I thought Object.assign was a es6 feature only, meaning we will probably need a polyfill of some sort. Wdyt?

File: src/MarkdownEditor.js [hunk 4]
```diff
@@ -78,15 +91,15 @@ var MarkdownEditor = React.createClass({
     };
 
     if(this.props.styles.styleMarkdownEditorContainer) {
-      styleMarkdownEditorContainer = this.props.styles.styleMarkdownEditorContainer;
+      Object.assign(styleMarkdownEditorContainer, this.props.styles.styleMarkdownEditorContainer);
     }
 
     return (
-      <div
+      <div id='container'
```
Comment (jrm2k6) [inline src/MarkdownEditor.js]: Can we use a classname instead of an id?

File: src/components/MarkdownEditorMenu.js [hunk 5]
```diff
@@ -1,8 +1,14 @@
...
+var Reflux = require('reflux');
+var ButtonManagerMixin = require('../mixins/ButtonManagerMixin');
+var MarkdownSelectionStore = require('../stores/MarkdownSelectionStore');
+
 var MarkdownEditorMenu = React.createClass({
   mixins: [Reflux.ListenerMixin, ButtonManagerMixin],
 
   propTypes: {
-    iconsSet: React.PropTypes.string.isRequired
+    iconsSet: React.PropTypes.string.isRequired,
```
Comment (jrm2k6) [inline src/components/MarkdownEditorMenu.js]: no need for the trailing comma

File: src/MarkdownEditor.js (hunk 3 above)
Comment (jkruse14) [inline src/MarkdownEditor.js]: yep, you're right, I forgot about that. I'm kind of green when it comes to javascript -- I'm a backend dev that's been living in Java for a long time. I'll look into that.

Comment (jkruse14) [inline src/MarkdownEditor.js]: Would it be best to use [babel-polyfill](https://babeljs.io/docs/usage/polyfill/), [babel-transform-object-assign](https://babeljs.io/docs/plugins/transform-object-assign/), or just create a 'polyfill' file in utils?

File: src/MarkdownEditor.js (hunk 4 above)
Comment (jkruse14) [inline src/MarkdownEditor.js]: whoops... this was a debugging item for my linked project to make sure I was working with the element I wanted. I'll remove

File: src/MarkdownEditor.js (hunk 3 above)
Comment (jrm2k6) [inline src/MarkdownEditor.js]: I usually use that one https://www.npmjs.com/package/object-assign

**Comment by jkruse14** [comment]
I will take a look at those units a little later today and follow that up with documentation updates

**Comment by jkruse14** [comment]
The error in the tests:
`SyntaxError: Unexpected token ]`

Is happening on master (without my changes). Is this a config issue somewhere?

**Comment by jrm2k6** [comment]
You are right, sorry for that. I will fix them before merging your changes. Could be related to jest versions or what not!

Just a bit of documentation and we are all clear.

File: README.md [hunk 6]
```diff
@@ -52,12 +52,75 @@ React.render(<TestComponent />, document.getElementById('content'));
...
+   - ```styles```, see [Styling](#styling) below
 
 You can also listen to content changes on the editor. If you are using Reflux, by listening to the changes on ```MarkdownEditorContentStore```.
 To be able to do so, just ```require('react-markdown-editor').MarkdownEditorContentStore;```
 
 ## Dependencies
 You can modify the styles directly by modifying the styles declared in ```dist/MarkdownEditor.js```. The pre-existing styles assume that you are using Bootstrap and Font Awesome.
+
+## Styling<a name="styling"></a>
+```<MarkdownEditor />``` is styled using React's inline styling and can be styled by supplying a ```styles``` prop. The following keys are used to style each component, below are their default values:
```
Comment (jrm2k6) [inline README.md]: Do you have an example of a customized editor? A screenshot would be enough to put in the README if you are ok with it!

**Pull request merged**